src/main/scala/org/fortysevendeg/sparksftp/ReadingSFTPConnectorApp.scala

```diff
@@ -28,47 +28,81 @@ object ReadingSFTPConnectorApp extends IOApp {
           config.spark.serializer.contains("KryoSerializer").toString
         )
         .set("spark.hadoop.fs.sftp.impl", "org.apache.hadoop.fs.sftp.SFTPFileSystem")
+        .set("spark.sql.legacy.allowCreatingManagedTableUsingNonemptyLocation","true")
         .registerKryoClasses(RegisterInKryo.classes.toArray)
 
       sparkSession: SparkSession = SparkSession.builder
         .config(defaultSparkConf)
         .enableHiveSupport
         .getOrCreate()
 
+      sftpUser1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_USER")
+        .getOrElse(config.sftp.sftpUser)
+      sftpPass1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_PASS")
+        .getOrElse(config.sftp.sftpPass)
+      sftpHost1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_HOST")
+        .getOrElse(config.sftp.sftpHost)
+      sftpPath1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_PATH")
+        .getOrElse(config.sftp.sftpPath)
+
+      _ = println(s"#### From SparkConf: $sftpUser1, $sftpHost1, $sftpPath1")
+
       // Construct Spark data frame reading a file from SFTP
       data: DataFrame = sparkSession.read
         .format("com.springml.spark.sftp") //TODO: Is this library single threaded?
```

Comment:
`//TODO: Is this library single threaded?`

Are we able to answer that question?

Reply:
Unfortunately, I have not done more experiments regarding paralellization yet. What I have done is that I have started experimenting with using DataStreamReaders and DataStreamWriters in another branch, to work on a different approach instead of batching. 
The prototype that uses springml does not allow the use of DataStreamReaders, and the second one with Hadoop support cannot read from SFTP when internally starts using a FileStreamSource.